### PR TITLE
DEV: Remove distutils

### PR DESCRIPTION
**Pull request by md-shafiul-alam**
### Description

distutils is deprectaed and is planned to be removed from python 3.12. This PR replaces the functionality used from distutils with setuptools and others.

---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.  
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

**Review by ethanglaser — APPROVED** [review]

File: scripts/build_backend.py [hunk 1]
```diff
@@ -16,18 +16,20 @@
...
 import sys
-from distutils import log
-from distutils.sysconfig import get_config_var, get_python_inc
 from math import floor
 from os.path import join as jp
+from sysconfig import get_config_var, get_paths
 
 import numpy as np
 
+logger = logging.getLogger("sklearnex")
```
Comment (samir-nasibli) [inline scripts/build_backend.py:31]: Just make sure that all logs works as expected, nothing missed.
Compare it with old one please.

**Review by samir-nasibli — COMMENTED** [review]
Thank you @md-shafiul-alam 
Looks good to me. Please rebase your branch

**Comment by samir-nasibli** [comment]
@md-shafiul-alam I found out that this cause issue on py<3.12 when packaging is not available in env

**Pull request merged**